src/sentry/db/models/manager.py

```diff
@@ -313,3 +321,27 @@ def post_delete(self, instance, **kwargs):
         """
         Triggered when a model bound to this manager is deleted.
         """
+
+
+class OrganizationBoundManager(BaseManager):
+    # TODO(dcramer): its not clear to me that we actually need to
+    # use BaseManager ever on related_fields (other this changes
+    # that behavior)
+    use_for_related_fields = False
```

Comment:
One concern here is we might actually want this in some cases. We wouldn't want it ever for ForeignKey's, but for a reverse mapper we would (e.g. organization.project_set should be constrained on team_id eventually).